internal/oci/uvm.go

```diff
@@ -228,6 +229,34 @@ func handleSecurityPolicy(ctx context.Context, a map[string]string, lopts *uvm.O
 	}
 }
 
+// handleWCOWSecurityPolicy handles parsing confidential pods related options and setting
+// implied options from the results for WCOW.
+func handleWCOWSecurityPolicy(ctx context.Context, a map[string]string, wopts *uvm.OptionsWCOW) error {
+	wopts.SecurityPolicy = ParseAnnotationsString(a, annotations.WCOWSecurityPolicy, wopts.SecurityPolicy)
+	if len(wopts.SecurityPolicy) == 0 {
+		return nil
+	}
+	wopts.SecurityPolicyEnabled = true
+
+	// overcommit isn't allowed when running in confidential mode and minimum of 2GB memory is required.
+	// We can change default values here, but if user provided specific values in annotations we should error out.
+	wopts.MemorySizeInMB = ParseAnnotationsUint64(ctx, a, annotations.MemorySizeInMB, 2048)
+	if wopts.MemorySizeInMB < 2048 {
+		return fmt.Errorf("minimum 2048MB of memory is required for confidential pods, got: %d", wopts.MemorySizeInMB)
+	}
+
+	wopts.AllowOvercommit = ParseAnnotationsBool(ctx, a, annotations.AllowOvercommit, false)
+	if wopts.AllowOvercommit {
+		return fmt.Errorf("allow overcommit MUST be false for confidential pods")
+	}
```

Comment:
`handleLCOWSecurityPolicy` silently overwrites `AllowOvercommit`, can we do the same here (and for `MemorySizeInMB` as well), and just warn instead?

Reply:
Not sure about LCOW, but for CWCOW I think the user should be made aware that the pod won't run if a lower memory is specified or if allow overcommit is set. We can log it, but the most users won't notice it from the logs and will keep thinking that the pod is running with a very low memory.